fees/SmarDex/index.ts

```diff
@@ -1,52 +1,107 @@
+
+// Methodology descriptions
+const FEES_METHODOLOGY = `
+A minor fee is collected on each swap, functioning as trading fees.
+The fees are set at 0.07% on Ethereum and 0.1% on other chains.
+On other networks, the fees may vary between different pairs and chains.
+Refer to https://docs.smardex.io/overview/what-is-smardex/fees for detailed information.
+`;
+
+const methodology = {
+  UserFees: FEES_METHODOLOGY,
+  Fees: FEES_METHODOLOGY,
+  Revenue: `0.02% of each swap on Ethereum is collected for the staking pool (SDEX holders that staked).`,
+  ProtocolRevenue: `Protocol has no revenue.`,
+  SupplySideRevenue: `0.05% of each swap on Ethereum is collected for the liquidity providers.`,
+  HoldersRevenue: `0.02% of each swap on Ethereum is collected for the staking pool (SDEX holders that staked).`,
 };
 
+// Define the adapter
+const adapter: Adapter = { adapter: {} };
 for (let chain in FEES) {
-  const { LP_FEES, POOL_FEES } = FEES[chain];
-  const TOTAL_FEES = LP_FEES + POOL_FEES;
-
-  // Convert fees to percentages and round to two decimal places
-  const totalFeesPercent = (TOTAL_FEES * 100).toFixed(2);
-  const lpFeesPercent = (LP_FEES * 100).toFixed(2);
-  const poolFeesPercent = (POOL_FEES * 100).toFixed(2);
-
-  const baseAdapter = getDexChainFees({
-    userFees: TOTAL_FEES,
-    totalFees: TOTAL_FEES,
-    supplySideRevenue: LP_FEES,
-    revenue: POOL_FEES,
-    holdersRevenue: POOL_FEES,
-    volumeAdapter,
-  });
-
-  const methodology = {
-    UserFees: `${totalFeesPercent}% of each swap is collected from the user that swaps as trading fees.`,
-    Fees: `${totalFeesPercent}% of each swap is collected from the user that swaps as trading fees.`,
-    Revenue: `${poolFeesPercent}% of each swap is collected for the staking pool (SDEX holders that staked).`,
-    ProtocolRevenue: `Protocol has no revenue.`,
-    SupplySideRevenue: `${lpFeesPercent}% of each swap is collected for the liquidity providers.`,
-    HoldersRevenue: `${poolFeesPercent}% of each swap is collected for the staking pool (SDEX holders that staked).`,
+  adapter.adapter[chain] = {
+    fetch: (timestamp: number) =>
+      feesFromSubgraph(timestamp, chain.toLocaleLowerCase()),
+    start: async () => CHAIN_STARTS[chain],
+    meta: { methodology },
   };
+}
 
-  adapter.adapter[chain] = {
-    ...baseAdapter[chain],
-    meta: {
-      methodology,
-    },
+/**
+ * Fetch fees from the subgraph for a given timestamp and chain.
+ *
+ * @param time - The timestamp to fetch fees at.
+ * @param chain - The blockchain chain.
+ * @returns Promise containing fetch results.
+ */
+export async function feesFromSubgraph(
+  time: number,
+  chain: string
+): Promise<FetchResult> {
+  const dayId = Math.floor(time / 86400);
+  const timestamp = getTimestampAtStartOfDayUTC(time);
+  const graphQuery = gql`
+    {
+      feeDayDatas(first: 1, where: { dayId_lte: ${dayId} }) {
+        dailyFeesPoolUSD
+        dailyFeesLPUSD
+        totalFeesPoolUSD
+        totalFeesLPUSD
```

Comment:
Typo on new naming ?
```suggestion
        dailyFeesPoolUSD
        dailyFeesLpUSD
        totalFeesPoolUSD
        totalFeesLpUSD
```

Reply:
Oh yes, exactly